### PR TITLE
Update OSX packaging for per-mod launchers.

### PR DESCRIPTION
**Pull request by pchote**
This updates the OSX packaging scripts to generate separate .app launchers for each mod and replaces the .zip wrapper with a proper disk image.

~~Depends on #13049 and~~ uses https://github.com/OpenRA/OpenRALauncherOSX/pull/2.

osx-dmg

Test build: https://www.dropbox.com/s/ut7qab548qewz3k/OpenRA-git-935e0fe.dmg?dl=0

### CODE REVIEW

**Review by phrohdoh — APPROVED** [review]
I won't be able to test these changes locally for a few days but the changes look sound, so feel free to merge without me testing. 👍

**Comment by pchote** [comment]
Rebased after #13049.

File: packaging/osx/buildpackage.sh [hunk 1]
```diff
@@ -14,24 +14,73 @@ if [ -e "OpenRA.app" ]; then
...
+}
+
+# Copies the game files and sets metadata
+populate_template() {
+	cp -r OpenRA.app "build/$1"
+	cp -r $BUILTDIR/* "build/$1/Contents/Resources/" || exit 3
+	cp "$2.icns" "build/$1/Contents/Resources/OpenRA.icns"
+	modify_plist "{MOD_ID}" "$2"  "build/$1/Contents/Info.plist"
+	modify_plist "{MOD_NAME}" "$3"  "build/$1/Contents/Info.plist"
+	modify_plist "{JOIN_SERVER_URL_SCHEME}" "openra-$2-$TAG"  "build/$1/Contents/Info.plist"
```
Comment (Phrohdoh) [inline packaging/osx/buildpackage.sh]: Nit: unnecessary spaces in these lines

**Comment by pchote** [comment]
Updated.

**Review by phrohdoh — APPROVED** [review]

**Comment by pchote** [comment]
I'm going to :+1: this myself, as we don't have anyone else with the expertise to give more than a token approval.

**Pull request merged**